### PR TITLE
Register thread names with OS

### PR DESCRIPTION
**Pull request by dcreager**
Each `cork_thread` instance already has a name, so we might as well register that name with the OS.

### CODE REVIEW

File: src/libcork/pthreads/thread.c [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+ * Copyright © 2013-2015, RedJack, LLC.
  * All rights reserved.
  *
  * Please see the COPYING file in this distribution for license details.
  * ----------------------------------------------------------------------
  */
 
+#if defined(__linux)
+/* This is needed on Linux to get the pthread_setname_np function. */
+#if !defined(_GNU_SOURCE)
```
Comment (johnzachary) [inline src/libcork/pthreads/thread.c]: Beautiful

File: src/libcork/pthreads/thread.c [hunk 2]
```diff
@@ -116,17 +124,30 @@ cork_thread_get_id(struct cork_thread *self)
...
     struct cork_thread_descriptor  *desc = cork_thread_descriptor_get();
+#if defined(__APPLE__) && defined(__MACH__)
+    char  thread_name[PTHREADS_MAX_THREAD_NAME_LENGTH];
+#endif
 
     desc->current_thread = self;
     desc->id = self->id;
     rc = self->run(self->user_data);
 
+#if defined(__APPLE__) && defined(__MACH__)
```
Comment (johnzachary) [inline src/libcork/pthreads/thread.c]: It's confusing to see the same (platform dependent) code in two separate functions, one a callback used in the other.

Comment (dcreager) [inline src/libcork/pthreads/thread.c]: Nevertheless it's needed — we have to set the name from outside the new thread on Linux, and from inside on OS X.

Comment (johnzachary) [inline src/libcork/pthreads/thread.c]: Can you add a note to that effect?

**Comment by johnzachary** [comment]
+1 to merge. Passes on OS X and Centos7.

**Comment by dcreager** [comment]
The Jenkins failure is because of a temporary DNS issue.

**Pull request merged**